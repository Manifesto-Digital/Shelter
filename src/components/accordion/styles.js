import styled from 'styled-components';
import SVG from 'react-inlinesvg';
import { buttonReset } from '../styled/buttons';

export const Wrapper = styled.div`
  position: relative;

  &:not(:first-of-type) {
    &:before {
      position: absolute;
      content: ' ';
      left: 0;
      top: ${({ theme }) => `-${theme.spacing.small}`};
      width: 100%;
      border-top: ${({ theme, active }) =>
        active && `1px solid ${theme.palette.grey45}`};
    }
  }
`;

export const Heading = styled.h3`
  position: relative;
  font-weight: normal;
  color: ${({ theme }) => theme.palette.grey10};
  margin-bottom: ${({ active }) => active && '0'};
`;

export const HeadingButton = styled.button`
  position: relative;
  ${buttonReset};
  padding: ${({ theme }) =>
    `${theme.spacing.standard} ${theme.spacing.xl} ${theme.spacing.standard} ${
      theme.spacing.standard
    }`};
  width: 100%;
  text-align: left;
  font-size: ${({ theme }) => theme.headers.h3};
`;

export const Content = styled.div`
  display: ${({ isOpen }) => (isOpen ? 'block' : 'none')};
  padding: ${({ theme }) =>
    `${theme.spacing.standard} ${theme.spacing.standard} 0 ${
      theme.spacing.standard
    }`};
`;

export const ArrowSVG = styled(SVG)`
  position: absolute;
  transform: ${({ isOpen }) => isOpen && 'rotate(180deg)'} translateY(-50%);
  transform-origin: top;
  top: 50%;
  right: ${({ theme }) => theme.spacing.standard};
  width: 20px;
  height: 25px;

  svg {
    display: block;
    width: 20px;
    height: 25px;
  }
`;
