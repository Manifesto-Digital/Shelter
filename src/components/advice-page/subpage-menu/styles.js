import styled from 'styled-components';

export const MenuNav = styled.nav`
  margin-bottom: ${({ theme }) => theme.spacing.large};

  a {
    text-decoration: none;
  }
`;
