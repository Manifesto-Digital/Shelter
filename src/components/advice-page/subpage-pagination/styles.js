import styled from 'styled-components';
import SVG from 'react-inlinesvg';

export const StyledLink = styled.a`
  text-decoration: none;
  margin-right: ${({ type }) => (type === 'prev' ? 'auto' : null)};
  margin-left: ${({ type }) => (type === 'next' ? 'auto' : null)};

  span {
    text-decoration: underline;
    font-weight: normal;
  }
`;

export const DirectionIcon = styled.div`
  display: flex;
  flex-direction: ${({ type }) => (type === 'prev' ? 'row-reverse' : 'row')};
  justify-content: flex-end;
  font-size: ${({ theme }) => theme.fontsize.larger};
  margin: 0 0 0.4em;
`;

export const Wrapper = styled.div`
  display: flex;
  margin-bottom: ${({ theme }) => theme.spacing.large};
  padding-bottom: ${({ theme }) => theme.spacing.large};
  border-bottom: 2px solid ${({ theme }) => theme.palette.offWhite};
`;

export const StyledSVG = styled(SVG)`
  svg {
    display: block;
    width: 35px;
    height: 40px;
  }
`;
