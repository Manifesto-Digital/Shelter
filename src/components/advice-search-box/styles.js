import styled from 'styled-components';
import Button from '../button';
import { breakpoint } from '../theme/breakpoint';
import TextInput from '../forms/text-input';

export const AdviceSearchForm = styled.form`
  padding: ${props => props.theme.spacing.standard};
  background-color: ${props => props.theme.palette.offWhite};
`;

export const SearchInput = styled(TextInput)`
  margin-bottom: 1em;
  ${breakpoint.tablet`
        display: inline-block;
        width: 75%;
        margin-right: 2em;
    `};
`;

export const SearchButton = styled(Button)`
  background: ${props => props.theme.palette.sanMarinoBlue};
  position: relative;
  width: auto;
  padding: ${props => props.theme.spacing.small};
  color: ${props => props.theme.palette.white};
  display: block;
  ${breakpoint.tablet`
        display: inline-block;
    `};
`;
