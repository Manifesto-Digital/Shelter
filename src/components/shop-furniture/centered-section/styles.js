import styled from 'styled-components';
import { breakpoint } from '../../theme/breakpoint';

export const Wrapper = styled.section`
  padding: ${props => props.theme.spacing.large} 0;
  background-color: ${props => props.theme.palette.grey10};
  margin-bottom: ${props => props.theme.spacing.large};
`;

export const Content = styled.div`
  text-align: center;
  margin: 0 auto;

  ${breakpoint.tablet`
    max-width: 66.666%
  `};

  & > p:last-of-type {
    margin-bottom: 0;
  }
`;

export const Red = styled.span`
  color: ${props => props.theme.palette.primary};
`;
