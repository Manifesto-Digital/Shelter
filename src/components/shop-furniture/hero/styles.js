import styled from 'styled-components';
import Image from '../../image';
import { breakpoint } from '../../theme/breakpoint';

export const Wrapper = styled.section`
  padding-top: ${({ theme }) => theme.spacing.large};
  margin-bottom: ${({ theme }) => theme.spacing.large};
`;
export const FlexWrapper = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;

  ${breakpoint.tablet`
    flex-wrap: nowrap;
  `};
`;

export const Content = styled.div`
  ${breakpoint.tablet`
    padding-right: ${({ theme }) => theme.spacing.standard};
  `};
`;
export const SubHeading = styled.p`
  font-size: ${({ theme }) => theme.headers.h3};

  span {
    color: ${({ theme }) => theme.palette.primary};
  }
`;

export const HeroImage = styled.div`
  flex-shrink: 0;
  ${breakpoint.tablet`
    width: 50%;
    padding-left: ${({ theme }) => theme.spacing.standard};
  `};
  ${breakpoint.desktop`
    width: 30%;
  `};
`;

export const StyledImage = styled(Image)`
  flex-shrink: 0;
`;
