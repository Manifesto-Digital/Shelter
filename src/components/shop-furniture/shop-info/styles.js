import styled from 'styled-components';
import SVG from 'react-inlinesvg';
import { breakpoint } from '../../theme/breakpoint';

export const Wrapper = styled.section`
  margin-bottom: ${props => props.theme.spacing.large};
  padding: ${props => props.theme.spacing.xl} 0;
  background-color: ${props => props.theme.palette.grey10};
`;

export const Flex = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
`;

export const Flex25 = styled.div`
  width: 100%;
  margin-bottom: ${props => props.theme.spacing.standard};
  background: ${props => props.theme.palette.white};

  &:last-of-type {
    margin-bottom: 0;
  }

  ${breakpoint.mobileLand`
    flex-basis: calc(50% - ${props => props.theme.spacing.small})
  `};

  ${breakpoint.desktop`
      flex: 1;
      width: auto;
      margin-right: ${props => props.theme.spacing.medium};
      margin-bottom: 0;
      
      &:last-of-type {
        margin-right: 0;
    `};
`;

export const Content = styled.div`
  padding: ${props => props.theme.spacing.standard};
  white-space: pre-line;

  p {
    margin-bottom: 0;
  }
`;

export const Heading = styled.h3`
  padding-bottom: ${props => props.theme.spacing.standard};
  border-bottom: 1px solid ${props => props.theme.palette.grey10};
  display: flex;
  align-items: center;
  font-size: ${props => props.theme.headers.h4};
`;

export const Icon = styled(SVG)`
  margin-right: ${props => props.theme.spacing.small};
  display: inline-block;
  height: 20px;
  width: 20px;

  svg {
    display: block;
    width: 20px;
    height: 20px;
  }
`;
