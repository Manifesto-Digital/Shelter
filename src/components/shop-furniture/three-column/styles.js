import styled from 'styled-components';
import SVG from 'react-inlinesvg';
import { breakpoint } from '../../theme/breakpoint';

export const Wrapper = styled.section`
  margin-bottom: ${props => props.theme.spacing.large};
`;

export const Flex = styled.div`
  display: flex;
  flex-wrap: wrap;
`;

export const Flex25 = styled.div`
  width: 100%;
  margin-bottom: ${props => props.theme.spacing.standard};

  ${breakpoint.mobileLand`
        width: 50%;
        padding-right: ${props => props.theme.spacing.standard};
    `};
  ${breakpoint.desktop`
        width: 25%;
        padding-right: ${props => props.theme.spacing.standard};
    `};
`;

export const Flex50 = styled.div`
  width: 100%;

  ${breakpoint.desktop`
        width: 50%;
        padding-left: ${props => props.theme.spacing.standard};
    `};
`;

export const UnorderedList = styled.ul`
  padding: 0;
  margin-bottom: 0;
`;

export const ListItem = styled.li`
  position: relative;
  list-style: none;
  position: relative;
  margin-bottom: 5px;
  padding-left: ${props => props.theme.spacing.standard};

  &:last-of-type {
    margin-bottom: 0;
  }
`;

export const ListIcon = styled(SVG)`
  margin-right: ${props => props.theme.spacing.small};
  position: absolute;
  top: 0;
  left: 0;

  svg {
    width: 10px;
  }

  path {
    fill: ${props => props.theme.palette.primary};
  }
`;
