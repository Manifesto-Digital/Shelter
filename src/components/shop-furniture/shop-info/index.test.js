import React from 'react';
import { shallow } from 'enzyme';
import { snapshotComponent } from '../../../../__tests__/helpers/index';
import { createFactory } from '../../../utils/test-factories';
import ShopInfo from '.';
import { Heading } from './styles';

// Default props
export const createShopInfo = createFactory({
  address: {
    childContentfulRichText: {
      html: '<p>51 Church RoadÂ \nLondon\nSW13 9HH</p>',
    },
  },
  openingHours: {
    childContentfulRichText: {
      html:
        '<p><strong>Mon to Sat</strong>: 10am to 6pm\n<strong>Sunday</strong>: 11am to 5pm</p>',
    },
  },
  parking: {
    childContentfulRichText: {
      html: '<p>No parking</p>',
    },
  },
  disabledAccess: {
    childContentfulRichText: {
      html: '<p>No disabled access</p>',
    },
  },
});

it('renders correctly', () => {
  const mockData = createShopInfo();
  snapshotComponent(<ShopInfo {...mockData} />);
});

it('renders the 4 shop information headings', () => {
  const mockData = createShopInfo();
  const wrapper = shallow(<ShopInfo {...mockData} />);
  const headingsToMatch = [
    'Address',
    'Opening hours',
    'Parking',
    'Disabled access',
  ];

  // Find all headings rendered and store them in an array
  const headings = wrapper.find(Heading).map(x => x.text());

  headingsToMatch.forEach(heading => {
    expect(headings.includes(heading)).toBe(true);
  });
});
