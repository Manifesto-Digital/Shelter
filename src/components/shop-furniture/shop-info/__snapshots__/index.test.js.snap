// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c7 ul {
  padding-left: 0;
  list-style: none;
}

.c7 ul li {
  position: relative;
  padding-left: 40px;
}

.c7 ul li::before {
  content: ' ';
  position: absolute;
  top: 8px;
  left: 10px;
  display: block;
  width: 10px;
  height: 10px;
  background: #212122;
}

.c7 ul li ul {
  padding-left: 0;
  list-style: none;
}

.c7 ul li ul li:before {
  background: transparent;
  border: 1px solid #212122;
}

.c7 ol li::before {
  min-width: 10px;
}

.c7 blockquote {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
  margin-right: 20px;
  margin-bottom: 0;
  padding: 60px 20px 20px 20px;
  border-left: 4px solid #ea232d;
  font-size: 1.35em;
  font-weight: 300;
}

.c7 blockquote:before {
  content: '';
  display: block;
  position: absolute;
  width: 30px;
  height: 30px;
  background-image: url(test-file-stub);
  top: 10px;
  left: 20px;
  color: #ea232d;
}

.c0 {
  margin-bottom: 40px;
  padding: 60px 0;
  background-color: #e5e5e5;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  width: 100%;
  margin-bottom: 20px;
  background: #fff;
}

.c3:last-of-type {
  margin-bottom: 0;
}

.c4 {
  padding: 20px;
  white-space: pre-line;
}

.c4 p {
  margin-bottom: 0;
}

.c5 {
  padding-bottom: 20px;
  border-bottom: 1px solid #e5e5e5;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1.125em;
}

.c6 {
  margin-right: 10px;
  display: inline-block;
  height: 20px;
  width: 20px;
}

.c6 svg {
  display: block;
  width: 20px;
  height: 20px;
}

.c1 {
  max-width: 1140px;
  margin: 0 auto;
  padding: 0 20px;
}

@media (min-width:32em) {
  .c3 {
    -webkit-flex-basis: calc(50% - 10px);
    -ms-flex-preferred-size: calc(50% - 10px);
    flex-basis: calc(50% - 10px);
  }
}

@media (min-width:62em) {
  .c3 {
    -webkit-flex: 1;
    -ms-flex: 1;
    flex: 1;
    width: auto;
    margin-right: 30px;
    margin-bottom: 0;
  }

  .c3:last-of-type {
    margin-right: 0;
  }
}

<section
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <div
        className="c3"
      >
        <div
          className="c4"
        >
          <h3
            className="c5"
          >
            <span
              className="isvg loading c6"
            />
            Address
          </h3>
          <div
            className="c7"
            dangerouslySetInnerHTML={
              Object {
                "__html": "<p>51 Church RoadÂ 
London
SW13 9HH</p>",
              }
            }
          />
        </div>
      </div>
      <div
        className="c3"
      >
        <div
          className="c4"
        >
          <h3
            className="c5"
          >
            <span
              className="isvg loading c6"
            />
            Opening hours
          </h3>
          <div
            className="c7"
            dangerouslySetInnerHTML={
              Object {
                "__html": "<p><strong>Mon to Sat</strong>: 10am to 6pm
<strong>Sunday</strong>: 11am to 5pm</p>",
              }
            }
          />
        </div>
      </div>
      <div
        className="c3"
      >
        <div
          className="c4"
        >
          <h3
            className="c5"
          >
            <span
              className="isvg loading c6"
            />
            Parking
          </h3>
          <div
            className="c7"
            dangerouslySetInnerHTML={
              Object {
                "__html": "<p>No parking</p>",
              }
            }
          />
        </div>
      </div>
      <div
        className="c3"
      >
        <div
          className="c4"
        >
          <h3
            className="c5"
          >
            <span
              className="isvg loading c6"
            />
            Disabled access
          </h3>
          <div
            className="c7"
            dangerouslySetInnerHTML={
              Object {
                "__html": "<p>No disabled access</p>",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
</section>
`;
